module1-introduction-to-backend/r3.4-api-best-practices/README.md

```diff
@@ -0,0 +1,232 @@
+# API Best Practices
+
+Now that we have learnt to build our own APIs, in this lesson we will cover some best practices for API development.
+
+Although some of these suggestions may require going through more modules of the course before implementation, consider this as a guide that you can come back and refer to at any time as you progress through the course.
+
+## What is API design?
+API design is the collection of planning and architectural decisions you make when building an API. Your basic API design influences how well developers are able to consume it and even how they use it. Just like website design or product design, API design informs the user experience. Good API design principles meet initial expectations and continue to behave consistently and predictably.
+
+Frontend developers need good UI design before building the product for their end users. Similarly backend developers need good API design because the users in this case are the frontend developers who will integrate with your API to build the end-user facing product.
+
+## Characteristics of a well-designed API
+In general, an effective API design will have the following characteristics:
+- **Easy to read and work with**: A well designed API will be easy to work with, and its resources and associated operations can quickly be memorized by developers who work with it constantly.
+- **Hard to misuse**: Implementing and integrating with an API with good design will be a straightforward process, and writing incorrect code will be a less likely outcome. It has informative feedback, and doesn’t enforce strict guidelines on the API’s end consumer.
+- **Complete and concise**: Finally, a complete API will make it possible for developers to make full-fledged applications against the data you expose. Completeness happens over time usually, and most API designers and developers incrementally build on top of existing APIs. It is an ideal which every engineer or company with an API must strive towards.
```

Comment:
i would add something in that it hides the implementation and developers need to know nothing about the implementation (aka has a clean interface)